### PR TITLE
feat!: remove `Course.position_in_program`

### PR DESCRIPTION
**Pull request by asadali145**
#### Pre-Flight checklist

- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
The first part of https://github.com/mitodl/mitxonline/issues/1220

#### What's this PR do?
Removes `Course.position_in_program`. We will be removing the `Course.program` in the follow-up PRs.

#### How should this be manually tested?

- Verify that the `Course.position_in_program` is removed.
- Verify that everything is working fine.
- Verify that the properties/methods removed are not used anywhere and are safely removed.

#### Any background context you want to provide?
As per the original ticket, it would be hard to manage the changes in a single PR. After a discussion, we decided to create multiple smaller PRs having similar kinds of changes. This is the first PR and it only removes changes related to `Course.position_in_program`.

### CODE REVIEW

**Review by jkachel — CHANGES_REQUESTED** [review]
One thing I noticed: the `configure_instance` command needs a small update to not specify `position_in_program`. Otherwise, seems to work OK for me 👍

**Review by rhysyngsun — COMMENTED** [review]
Generally LGTM, although could you remove the references to `--program-position` in `docs/source/commands/create_courseware.rst`?

**Review by jkachel — CHANGES_REQUESTED** [review]
I'm getting a test failure in `courses/serializers_test.py::test_serialize_program[False]`. Functionality looks OK though and `configure_instance` works now too.

**Comment by asadali145** [comment]
@jkachel OR @rhysyngsun Could you guys please have one final look and approve this? We will merge this PR and will create follow up Tickets/PR for the removal of `Course.program`.

File: courses/serializers.py [hunk 1]
```diff
@@ -516,52 +517,6 @@ class Meta:
         fields = ["uuid", "link"]
 
 
-class ProgramEnrollmentSerializer(serializers.ModelSerializer):
```
[the remark this answers is not in the document]
Comment (asadali145) [reply courses/serializers.py:519]: It was unused.

**Review by jkachel — APPROVED** [review]
Looks OK to me. I'm not sure why the `ProgramEnrollmentSerializer` is being removed but I'm not necessarily opposed to it being removed.

File: courses/serializers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rhysyngsun) [reply courses/serializers.py:519]: Makes sense, we shouldn't have unused code like that and if we want it back it's easy enough to bring back from git history.

**Pull request merged**